Lib/test/test_logging.py

```diff
@@ -6148,6 +6148,8 @@ def test_should_not_rollover(self):
 
     def test_should_rollover(self):
         rh = logging.handlers.RotatingFileHandler(self.fn, encoding="utf-8", maxBytes=1)
+        self.assertFalse(rh.shouldRollover(self.next_rec()))
+        rh.emit(self.next_rec())
```

Comment:
`self.next_rec()` returns a new record. You should save it and pass the same record to `shouldRollover()` and `emit()`.